processor/eqReceiptProcessor.go

```diff
@@ -0,0 +1,52 @@
+package processor
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/ONSdigital/census-rm-pubsub-adapter/config"
+	"github.com/ONSdigital/census-rm-pubsub-adapter/models"
+	"github.com/pkg/errors"
+)
+
+type EqReceiptProcessor struct {
+	*Processor
+}
+
+func NewEqReceiptProcessor(ctx context.Context, appConfig *config.Configuration) *EqReceiptProcessor {
+	eqReceiptProcessor := &EqReceiptProcessor{}
+	eqReceiptProcessor.Processor = NewProcessor(ctx, appConfig, appConfig.EqReceiptProject, appConfig.EqReceiptSubscription, appConfig.ReceiptRoutingKey, convertEqReceiptToRmMessage, unmarshalEqReceipt)
+	return eqReceiptProcessor
+}
+
+func unmarshalEqReceipt(data []byte) (models.PubSubMessage, error) {
+	var eqReceipt models.EqReceipt
+	err := json.Unmarshal(data, &eqReceipt)
+	if err != nil {
+		return nil, err
+	}
+	return eqReceipt, nil
+}
+
+func convertEqReceiptToRmMessage(receipt models.PubSubMessage) (*models.RmMessage, error) {
```

Comment:
mentioned this lower down... not sure if passing structs by value is best practice, given that it'll have to make a copy of it